### PR TITLE
Fix #38 add default path and version

### LINKED ISSUE #38: Add defaults when kubeclient is initialized
Based on:
https://github.com/ManageIQ/manageiq/pull/2156#issuecomment-81717931

Add defaults if they weren't passed from client:
1. /api suffix
2. apiversion

### PR DESCRIPTION
**Pull request by abonas**

### CODE REVIEW

**Comment by abonas** [comment]
@simon3z  / @Fryguy   pls review.
@Fryguy  -  this is a first step to pass less info on k8s provider side in manageiq when initializing the client

File: lib/kubeclient.rb [hunk 1]
```diff
@@ -22,8 +22,12 @@ class Client
       [Kubeclient.const_set(et, Class.new(RecursiveOpenStruct)), et]
     end
 
-    def initialize(uri, version)
+    def initialize(uri, version = 'v1beta3')
       @api_endpoint = (uri.is_a? URI) ? uri : URI.parse(uri)
+
+      unless @api_endpoint.path.include? 'api'
+        @api_endpoint.merge!(File.join(@api_endpoint.path, 'api'))
```
Comment (tenderlove) [inline lib/kubeclient.rb]: You probably don't want this because `File.join` will use the default file separator which is `\` on Windows.  I would just use string interpolation.

Comment (abonas) [inline lib/kubeclient.rb]: > You probably don't want this because File.join will use the default file separator which is \ on Windows. I would just use string interpolation.

good point, thanks !
I assume the same goes for line 31 too (which is not part of this patch though)

BTW from looking at the documentation of the method it's really unclear...

Comment (Fryguy) [inline lib/kubeclient.rb]: Actually since you have a URI object,  perhaps you can just use it,  which
would be the most accurate.

Comment (abonas) [inline lib/kubeclient.rb]: > Actually since you have a URI object, perhaps you can just use it, which would be the most accurate.

what do you mean?
user could just pass host:port as part of the uri, without "/api" and without the version.
in the case "/api" and version weren't passed, they need to be appended to the path in uri, no?

Comment (Fryguy) [inline lib/kubeclient.rb]: I guess I meant I don't think you should append.  I think you should only
default to /api if they didn't pass a path at all.   If they passed a
path,  then that is the caller stating they know better, and they have
declared a different endpoint.

Comment (abonas) [inline lib/kubeclient.rb]: > I guess I meant I don't think you should append. I think you should only default to /api if they didn't pass a path at all. If they passed a path, then that is the caller stating they know better, and they have declared a different endpoint.

ok now I get it... set path to "/api" if path is empty. makes sense!

File: test/test_kubeclient.rb [hunk 2]
```diff
@@ -20,6 +20,37 @@ def test_json
     assert_equal(expected, JSON.parse(JSON.dump(our_object.to_h)))
   end
 
+  def test_pass_uri
+    stub_request(:get, /\/services/)
+      .to_return(body: open_test_json_file('entity_list_b1.json'),
+                 status: 200)
+    uri = URI::HTTP.build(port: 8080)
```
Comment (simon3z) [inline test/test_kubeclient.rb]: No need for lines 24-26 if your assert is the only assert is line 33
